frontend/src/pages/InvitePage/InvitePage.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useGetIdentity, useNotify, useTranslate } from 'react-admin';
+import { useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useOutbox } from '@semapps/activitypub-components';
+import { Box, CircularProgress } from '@mui/material';
+import InvitePageViewLoggedOut from './InvitePageViewLoggedOut';
+import InvitePageViewLoggedIn from './InvitePageViewLoggedIn';
+import InvitePageSuccess from './InvitePageSuccess';
+import ChoosePodProviderPage from './InvitePageProviderSelect';
+import SimpleBox from '../../layout/SimpleBox';
+
+/** The URI is expected to be encoded in the URI fragment in the SearchParams format. */
+const getCapabilityUri = (location: Location) => {
+  // const searchFragment = location.toString().match('#(.*)')?.[1];
+  // if (!searchFragment) {
+  //   return undefined;
+  // }
```

Comment:
You did not use route params because of problems or because you think search params are better ? (I think we talked about this but I don't remember). Personnally I think that invite links like `http://localhost:5000/invite/http%3A%2F%2Flocalhost%3A3000%2Ftest4%2Fdata%2Fcapabilities%2Fa106a291-d1fb-44e6-aac3-4ad1b094cc6a` would be nicer because they would look more secret. :)

Reply:
I'm thinking now we could even base64 encode them (with URL-safe option) for more secret ! This would make them look like Telegram invitation links:  `https://localhost:5000/invite/aHR0cCUzQSUyRiUyRmxvY2FsaG9zdCUzQTMwMDAlMkZ0ZXN0NCUyRmRhdGElMkZjYXBhYmlsaXRpZXMlMkZhMTA2YTI5MS1kMWZiLTQ0ZTYtYWFjMy00YWQxYjA5NGNjNmE` :laughing:

Reply:
I disposed of the idea at that time because I think I was having trouble with the hash fragments when redirecting to / from the signup/login form.
But looking at the code right now, I can't really tell why. On the other hand it doesn't really improve security either since refer spoofing is not an issue and the login form will need to have the redirect URI with the capability encoded anyways - unless we want the login provider to be oblivious of the invite cap. In that case we would have to add custom logic and add the whole redirect URI in a hash fragment and make the login/signup page use that for redirecting. So I thought it wasn't quite worth the effort but it shouldn't be much to change it too, I suppose from a general point of view it would be a good practice.
I have a feeling that going with `http://localhost:5000/invite/<cap URI>` is semantically not so exact since the cap is used as a key not as a resource?

Regarding the base64 encoding, it would definitely look cool but I'm not sure if it conveys a false sense of secrecy. As a tech-user I'd probably be quite curios to see that the capability is just a URI, I can dereference. Most users would probably not care really.
But I agree that wrapping the cap URI with `encodeURIComponent` is a good idea :)

What do you think?

Reply:
> I have a feeling that going with `http://localhost:5000/invite/<cap URI>` is semantically not so exact since the cap is used as a key not as a resource?

IMO the semantic meaning of frontend URL is not so important, but anyway it makes sense to me: the capability URL is some kind of invitation in itself ("Hey, I give you the capability to do that"). In the future, we could even consider using the same URL for other kind of capabilities, like events invitation... ?

> Regarding the base64 encoding, it would definitely look cool but I'm not sure if it conveys a false sense of secrecy. As a tech-user I'd probably be quite curios to see that the capability is just a URI, I can dereference. Most users would probably not care really.

I agree with this false sense of secrecy ;)

Reply:
> IMO the semantic meaning of frontend URL is not so important, but anyway it makes sense to me: the capability URL is some kind of invitation in itself ("Hey, I give you the capability to do that"). 

So how would you go on with this? :)

> In the future, we could even consider using the same URL for other kind of capabilities, like events invitation... ?

Do you mean the `/invite` URI or to resolve the cap URI accordingly when requested with an accept header of `text/html` or so?

Reply:
> So how would you go on with this? :)

I would go with `http://localhost:5000/invite/http%3A%2F%2Flocalhost%3A3000%2Ftest4%2Fdata%2Fcapabilities%2Fa106a291-d1fb-44e6-aac3-4ad1b094cc6a` if this is not too complicated to implement.

> Do you mean the `/invite` URI or to resolve the cap URI accordingly when requested with an accept header of `text/html` or so?

Yes the `/invite` frontend URI. SemApps has a feature to redirect resources URIS to frontend routes (https://semapps.org/docs/middleware/ldp/#redirecting-to-a-frontend-app) but I don't think this is adapted for Pods.